check_output fails with stderr set to subprocess.STDOUT

This is how I am invoking a process in the code I am testing:

```
        output = subprocess.check_output(
            '/usr/lib/update-notifier/apt-check',
            stderr=subprocess.STDOUT).decode('utf8')
```

This is how I am registering the process:

```
    fake_process.register_subprocess('/usr/lib/update-notifier/apt-check',
                                     stdout=("0;0",))
```

This is what I get when I try to run my test:

```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
client/plugins/os_updates.py:255: in ubuntu_checker
    output = subprocess.check_output(
/usr/lib/python3.8/subprocess.py:411: in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
/usr/lib/python3.8/subprocess.py:489: in run
    with Popen(*popenargs, **kwargs) as process:
../../.virtualenvs/PenguinDome/lib/python3.8/site-packages/pytest_subprocess/core.py:262: in dispatch
    result.configure(**kwargs)
../../.virtualenvs/PenguinDome/lib/python3.8/site-packages/pytest_subprocess/core.py:126: in configure
    self.stdout = self._prepare_buffer(self.__stderr, self.stdout)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pytest_subprocess.core.FakePopen object at 0x7f4608c8cdc0>, input = None
io_base = <_io.BytesIO object at 0x7f46081dc4f0>

    def _prepare_buffer(self, input, io_base=None):
        linesep = self._convert(os.linesep)
    
        if isinstance(input, (list, tuple)):
            input = linesep.join(map(self._convert, input))
    
        if isinstance(input, str) and not self.text_mode:
            input = input.encode()
    
        if isinstance(input, bytes) and self.text_mode:
            input = input.decode()
    
        if input:
            if not input.endswith(linesep):
                input += linesep
            if self.text_mode and self.__universal_newlines:
                input = input.replace("\r\n", "\n")
    
        if io_base is not None:
>           input = io_base.getvalue() + input
E           TypeError: can't concat NoneType to bytes

../../.virtualenvs/PenguinDome/lib/python3.8/site-packages/pytest_subprocess/core.py:149: TypeError
```

--- a/changelog.d/bug.7bfb3945.entry.yaml
+++ b/changelog.d/bug.7bfb3945.entry.yaml
@@ -1,0 +1,8 @@
+git_email: github@aklajnert.pl
+git_user: Andrzej Klajnert
+message: Fixed crash for stderr redirect with an empty stream definition.
+os_user: andrzejk
+pr_ids:
+- '19'
+timestamp: 1587964114
+type: bug

--- a/pytest_subprocess/core.py
+++ b/pytest_subprocess/core.py
@@ -122,7 +122,7 @@ class FakePopen:
         if kwargs.get("stdout") == subprocess.PIPE:
             self.stdout = self._prepare_buffer(self.__stdout)
         stderr = kwargs.get("stderr")
-        if stderr == subprocess.STDOUT:
+        if stderr == subprocess.STDOUT and self.__stderr:
             self.stdout = self._prepare_buffer(self.__stderr, self.stdout)
         elif stderr == subprocess.PIPE:
             self.stderr = self._prepare_buffer(self.__stderr)
@@ -146,7 +146,7 @@ class FakePopen:
                 input = input.replace("\r\n", "\n")
 
         if io_base is not None:
-            input = io_base.getvalue() + input
+            input = io_base.getvalue() + (input)
 
         io_base = io.StringIO() if self.text_mode else io.BytesIO()
 

--- a/tests/test_subprocess.py
+++ b/tests/test_subprocess.py
@@ -688,3 +688,31 @@ def test_callback_with_arguments(fake_process):
     process.wait()
 
     assert process.returncode == return_code
+
+
+def test_subprocess_pipe_without_stream_definition(fake_process):
+    """
+    From GitHub #17 - the fake_subprocess was crashing if the subprocess was called
+    with stderr=subprocess.PIPE but the stderr was not defined during the process
+    registration.
+    """
+    fake_process.register_subprocess(
+        ["test-no-stderr"], stdout="test",
+    )
+    fake_process.register_subprocess(
+        ["test-no-stdout"], stderr="test",
+    )
+    fake_process.register_subprocess(["test-no-streams"],)
+
+    assert (
+        subprocess.check_output(["test-no-stderr"], stderr=subprocess.STDOUT).decode()
+        == "test" + os.linesep
+    )
+    assert (
+        subprocess.check_output(["test-no-stdout"], stderr=subprocess.STDOUT).decode()
+        == "test" + os.linesep
+    )
+    assert (
+        subprocess.check_output(["test-no-streams"], stderr=subprocess.STDOUT).decode()
+        == ""
+    )

